packages/rulesets/src/spectral/functions/resource-name-restriction.ts

```diff
@@ -10,14 +10,22 @@ export const resourceNameRestriction = (paths: any, _opts: any, ctx: any) => {
 
   function getPathParameter(pathItem: any, paramName: string) {
     let parameters: any[] = []
-    const method = Object.keys(pathItem).find((k) => k !== "parameters")
-    if (method) {
+    // Get all the methods defined 
+    const methods = Object.keys(pathItem).filter((k) => k !== "parameters")
+
+    for (const method of methods) {
       const operationParameters = pathItem[method].parameters
-      parameters = parameters.concat(operationParameters)
+      // Add parameters if exists
+      if (operationParameters) {
+        parameters = parameters.concat(operationParameters)
```

Comment:
Are there any cases where a parameter might be null itself? Or are there any cases where a parameter might not have an "in" or "name" property defined which would case the return statement to fail?